cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -63,104 +63,109 @@ void Run() {
   std::unordered_set<string> expected_sensor_ids;
 
   // For 2D SLAM, subscribe to exactly one horizontal laser.
-  ::ros::Subscriber laser_scan_subscriber;
+  rclcpp::Subscription<sensor_msgs::msg::LaserScan>::SharedPtr laser_scan_subscriber;
   if (options.use_laser_scan) {
-    laser_scan_subscriber = node.node_handle()->subscribe(
-        kLaserScanTopic, kInfiniteSubscriberQueueSize,
-        boost::function<void(const sensor_msgs::LaserScan::ConstPtr&)>(
-            [&](const sensor_msgs::LaserScan::ConstPtr& msg) {
+    laser_scan_subscriber = node.node_handle()->create_subscription<sensor_msgs::msg::LaserScan>(
+        kLaserScanTopic,
+        [&](sensor_msgs::msg::LaserScan::ConstSharedPtr msg) {
               node.map_builder_bridge()
                   ->sensor_bridge(trajectory_id)
                   ->HandleLaserScanMessage(kLaserScanTopic, msg);
-            }));
+        }, rmw_qos_profile_default);
```

Comment:
I'd guess you want to use sensor_data profile by befaut during your initial port and iterate on it later